### DISCUSSION TITLE
I don't get error or data from use mutation after executing

### DISCUSSION THREAD

**Discussion by abdelrazzaq-dev** [dotansimha/graphql-code-generator#4769] [General] [answered]
Hi, I try to get the error or data after execute the use mutation but I get undefined always in both variables data and error though there is a error I catch it.
according to the Apollo-client documentation the use mutation return an function to execute and an object that contain data, error, and loading but the generated function only return loading.
I can see the error in the apollo-link-error!
```ts
const [login, { loading, data, error }] = useLoginMutation();
onSubmit={async (values, { setSubmitting, setStatus }) => {
  setSubmitting(false);
  const variables = { key: values.key, password: values.password };
  try {
    const response = await login({ variables });
    if (response && response.data) {
      history.push('/');
    } else if (response.errors) {
      setStatus(response.errors);
    }
  } catch (err) {
    console.log(error); // undefined
    console.log(err);
    setStatus(error);
  }
}}
```

**Comment by dotansimha** [comment]
@AbderrazzakB can you please share a reproduction of that issue? the generated signature of `typescript-react-apollo` is just a wrapper for apollo-client, it doesn't change any functionality other than TS types.

**Comment by abdelrazzaq-dev** [comment, marked as answer]
Hi @dotansimha, OK I found out what was wrong, not sure why or how but I think when there is an error my mutation throw an error and it did reach my catch error before the mutation error property gets updated. because in the render I can see the error and display the error message from it.
